tests/components/islamic_prayer_times/test_init.py

```diff
@@ -0,0 +1,162 @@
+"""Tests for Islamic Prayer Times init."""
+
+from datetime import datetime, timedelta
+from unittest.mock import patch
+
+from prayer_times_calculator.exceptions import InvalidResponseError
+import pytest
+
+from homeassistant.components import islamic_prayer_times
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.setup import async_setup_component
+import homeassistant.util.dt as dt_util
+
+from tests.common import MockConfigEntry, async_fire_time_changed, mock_coro
+
+MOCK_OPTIONS = {islamic_prayer_times.CONF_CALC_METHOD: "makkah"}
+
+PRAYER_TIMES = {
+    "Fajr": "06:10",
+    "Sunrise": "07:25",
+    "Dhuhr": "12:30",
+    "Asr": "15:32",
+    "Maghrib": "17:35",
+    "Isha": "18:53",
+    "Midnight": "00:45",
+}
+
+
+async def test_setup_with_no_config(hass):
+    """Test that we do not discover anything or try to set up a Islamic Prayer Times."""
+    assert await async_setup_component(hass, islamic_prayer_times.DOMAIN, {}) is True
+    assert islamic_prayer_times.DOMAIN not in hass.data
+
+
+async def test_setup_with_config(hass):
+    """Test that we import the config and setup the client."""
+    config = {
+        islamic_prayer_times.DOMAIN: {islamic_prayer_times.CONF_CALC_METHOD: "isna"}
+    }
+    with patch(
+        "prayer_times_calculator.PrayerTimesCalculator.fetch_prayer_times",
+        return_value=PRAYER_TIMES,
+    ):
+        assert (
+            await async_setup_component(hass, islamic_prayer_times.DOMAIN, config)
+            is True
+        )
+
+
+async def test_successful_config_entry(hass):
+    """Test that Islamic Prayer Times is configured successfully."""
+
+    entry = MockConfigEntry(domain=islamic_prayer_times.DOMAIN, data={},)
+    entry.add_to_hass(hass)
+
+    with patch(
+        "prayer_times_calculator.PrayerTimesCalculator.fetch_prayer_times",
+        return_value=PRAYER_TIMES,
+    ):
+        assert await hass.config_entries.async_setup(entry.entry_id) is True
+        await hass.async_block_till_done()
+        assert entry.options == {
+            islamic_prayer_times.CONF_CALC_METHOD: islamic_prayer_times.DEFAULT_CALC_METHOD
+        }
+
+
+async def test_setup_failed(hass):
+    """Test Islamic Prayer Times failed due to an error."""
+
+    entry = MockConfigEntry(domain=islamic_prayer_times.DOMAIN, data={},)
+    entry.add_to_hass(hass)
+
+    # test request error raising ConfigEntryNotReady
+    with patch(
+        "prayer_times_calculator.PrayerTimesCalculator.fetch_prayer_times",
+        side_effect=InvalidResponseError(),
+    ), pytest.raises(ConfigEntryNotReady):
+
+        await islamic_prayer_times.async_setup_entry(hass, entry)
```

Comment:
Always use the core interfaces to start the tests. Use `hass.config_entries.async_setup`. Then assert the state of the entry. It will be `ENTRY_STATE_SETUP_RETRY`.